go/vt/vtctl/grpcvtctldserver/server_test.go

```diff
@@ -8914,3 +8915,681 @@ func TestValidateShard(t *testing.T) {
+		})
+	}
+}
+
+func TestValidateVSchema(t *testing.T) {
+	ctx := context.Background()
+
+	tests := []struct {
+		name       string
+		expVSchema *vschemapb.Keyspace
+		// mapping of shard -> primarySchema
+		schemaMap map[string]*tabletmanagerdatapb.SchemaDefinition
+		cell      string
+		tmc       *testutil.TabletManagerClient
+		ts        *topo.Server
+		req       *vtctldata.ValidateVSchemaRequest
+		shouldErr bool
+	}{
+		{
+			name: "should pass",
+			schemaMap: map[string]*tabletmanagerdatapb.SchemaDefinition{
+				"-80": {
+					TableDefinitions: []*tabletmanagerdatapb.TableDefinition{
+						{
+							Name:    "t1",
+							Columns: []string{"c1"},
+						},
+					},
+				},
+			},
+			expVSchema: &vschemapb.Keyspace{
+				Sharded: true,
+				Tables: map[string]*vschemapb.Table{
+					"t1": {
+						Columns: []*vschemapb.Column{
+							{
+								Name: "c1",
+								Type: querypb.Type_BINARY,
+							},
+						},
+						ColumnVindexes: []*vschemapb.ColumnVindex{
+							{
+								Column: "c1",
+								Name:   "c1",
+							},
+						},
+					},
+				},
+				Vindexes: map[string]*vschemapb.Vindex{
+					"c1": {
+						Type: "hash",
+					},
+				},
+			},
+			cell: "zone1",
+			ts:   memorytopo.NewServer("zone1"),
+			tmc: &testutil.TabletManagerClient{
+				GetSchemaResults: map[string]struct {
+					Schema *tabletmanagerdatapb.SchemaDefinition
+					Error  error
+				}{},
+			},
+			req: &vtctldatapb.ValidateVSchemaRequest{
+				Keyspace:      "ks",
+				Shards:        []string{"-80"},
+				ExcludeTables: nil,
+				IncludeViews:  true,
+			},
+			shouldErr: false,
+		},
+		{
+			name: "primary has different schema",
+			schemaMap: map[string]*tabletmanagerdatapb.SchemaDefinition{
+				"80-": {
+					TableDefinitions: []*tabletmanagerdatapb.TableDefinition{
+						{
+							Name:    "t2",
+							Columns: []string{"c1"},
+						},
+					},
+				},
+			},
+			expVSchema: &vschemapb.Keyspace{
+				Sharded: true,
+				Tables: map[string]*vschemapb.Table{
+					"t1": {
+						Columns: []*vschemapb.Column{
+							{
+								Name: "c1",
+								Type: querypb.Type_BINARY,
+							},
+						},
+						ColumnVindexes: []*vschemapb.ColumnVindex{
+							{
+								Column: "c1",
+								Name:   "c1",
+							},
+						},
+					},
+				},
+				Vindexes: map[string]*vschemapb.Vindex{
+					"c1": {
+						Type: "hash",
+					},
+				},
+			},
+			cell: "zone1",
+			ts:   memorytopo.NewServer("zone1"),
+			tmc: &testutil.TabletManagerClient{
+				GetSchemaResults: map[string]struct {
+					Schema *tabletmanagerdatapb.SchemaDefinition
+					Error  error
+				}{},
+			},
+			req: &vtctldatapb.ValidateVSchemaRequest{
+				Keyspace:      "ks",
+				Shards:        []string{"80-"},
+				ExcludeTables: nil,
+				IncludeViews:  true,
+			},
+			shouldErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			// Test Setup
+			tt.tmc.TopoServer = tt.ts
```

Comment:
applies to the other 2 tests as well .. do we actually need this? does the test not pass if you remove it?